packages/core/src/Synchronizer.ts

```diff
@@ -442,17 +466,46 @@ export class Synchronizer extends EventEmitter {
 
     async newGSTLeaf(event: ethers.Event, db: TransactionDB) {
         const epoch = Number(event.topics[1])
-        const leaf = BigInt(event.topics[2])
+        const hash = BigInt(event.topics[2]).toString()
         const index = Number(event.topics[3])
 
         db.create('GSTLeaf', {
             epoch,
-            hash: leaf.toString(),
+            hash,
             index,
         })
         return true
     }
 
+    async epochTreeLeaf(event: ethers.Event, db: TransactionDB) {
+        const epoch = Number(event.topics[1])
+        const leaf = BigInt(event.topics[2]).toString()
+        const index = BigInt(event.topics[3]).toString()
+
+        const exist = await this._db.findOne('EpochTreeLeaf', {
+            where: { epoch, index },
+        })
+        if (exist) {
```

Comment:
You can use an upsert operation here:
```
db.upsert('EpochTreeLeaf', {
  where: {
    epoch,
    index,
  },
  update: {
    leaf,
  },
  create: {
    epoch,
    index,
    leaf,
  }
})
```
This will be accelerated by the database when supported.

Reply:
It is weird that it will cause error `Error processing event: Error: anondb error: Error: SQLITE_ERROR: near ")": syntax error`